### PR TITLE
Indexed paylaod attestation test

### PR DESCRIPTION
**Pull request by Redidacove**
What does this Pr do ?

Adds Test for IndexedPayloadAttestation

### CODE REVIEW

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+func TestGetAttestingIndices(t *testing.T) {
+	attestingIndices := []primitives.ValidatorIndex{1, 2, 3}
+	nilAttestingIndices := []primitives.ValidatorIndex{}
+	pa := &IndexedPayloadAttestation{
+		AttestingIndices: attestingIndices,
+	}
+	got := pa.GetAttestingIndices()
+	for i, v := range got {
+		require.Equal(t, attestingIndices[i], v)
+	}
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:20]:
```suggestion
	require.DeepEqual(t, attestingIndices, got)
```

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	eth "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/testing/require"
+)
+
+func TestGetAttestingIndices(t *testing.T) {
+	attestingIndices := []primitives.ValidatorIndex{1, 2, 3}
+	nilAttestingIndices := []primitives.ValidatorIndex{}
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:13]: This is an empty slice not nil slice. It's a non-nil and empty slice. Calling it 'nil' is confusing at least to me.

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	}
+	got = pa.GetAttestingIndices()
+	require.DeepEqual(t, nilAttestingIndices, got)
+}
+
+func TestGetData(t *testing.T) {
+	data := &eth.PayloadAttestationData{
+		Slot: 1,
+	}
+	nilData := &eth.PayloadAttestationData{}
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:32]: Same here. It's not nil. It has a nil slice for BeaconBlockRoot and 0 for Slot and PayloadStatus respectively, which is their default value.

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	pa = &IndexedPayloadAttestation{
+		Data: nilData,
+	}
+	got = pa.GetData()
+	require.Equal(t, nilData, got)
+}
+
+func TestGetSignature(t *testing.T) {
+	sig := []byte{2}
+	nilSig := []byte{}
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:48]: Here also. It's a non-nil and empty byte slice. If you want a nil byte slice, you can do this: `var nilSig []byte`.

**Review by jihoonsong — COMMENTED** [review]
I think adding more test cases could make it more thorough.

File: consensus-types/epbs/indexed_payload_attestation_test.go (hunk 2 above)
Comment (jihoonsong) [reply consensus-types/epbs/indexed_payload_attestation_test.go:13]: If you really want a nil slice, you can do `nilAttestingIndices := []primitives.ValidatorIndex(nil)`

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+func TestGetAttestingIndices(t *testing.T) {
+	attestingIndices := []primitives.ValidatorIndex{1, 2, 3}
+	
+	nilAttestingIndices := []primitives.ValidatorIndex(nil)
+	pa := &IndexedPayloadAttestation{
+		AttestingIndices: attestingIndices,
+	}
+	got := pa.GetAttestingIndices()
+	require.DeepEqual(t, attestingIndices, got)
+	pa = nil
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:20]:
```suggestion
	pa = &IndexedPayloadAttestation{
		AttestingIndices: nilAttestingIndices,
	}
```

Comment (jihoonsong) [reply consensus-types/epbs/indexed_payload_attestation_test.go:20]: I think this is what you need to do. You can also declare an array of structs each of which corresponds to a test case. For example, you can refer to [this](https://github.com/prysmaticlabs/prysm/blob/develop/api/client/event/utils_test.go#L12).

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	testCases := []struct {
+		name string
+		sig  []byte
+	}{
+		{
+			name: "Test 1",
+			sig:  []byte{1, 2},
+		},
+		{
+			name: "Test 1",
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:93]:
```suggestion
			name: "Test 2",
```

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+			sig:  []byte{29, 100},
+		},
+		{
+			name: "Zero slot",
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:97]:
```suggestion
			name: "Zero signature",
```

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+			},
+		},
+		{
+			name: "Zero slot",
+			data: &eth.PayloadAttestationData{
+				Slot: primitives.Slot(0),
+			},
+		},
+		{
+			name: "Nil AttestingIndices",
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:68]:
```suggestion
			name: "Nil slot",
```
slot seems line up better with the other description.

File: consensus-types/epbs/indexed_payload_attestation_test.go [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+		name             string
+		attestingIndices []primitives.ValidatorIndex
+	}{
+		{
+			name:             "Test 1",
+			attestingIndices: []primitives.ValidatorIndex{1, 2, 3},
+		},
+		{
+			name:             "Test 2",
+			attestingIndices: []primitives.ValidatorIndex{55,66,787},
```
Comment (jihoonsong) [inline consensus-types/epbs/indexed_payload_attestation_test.go:22]:
```suggestion
			attestingIndices: []primitives.ValidatorIndex{55, 66, 787},
```
Linter.

**Review by jihoonsong — APPROVED** [review]
LGTM

**Review by potuz — APPROVED** [review]

**Pull request merged**